### DISCUSSION TITLE
Add a global configuration to allow fake PUT / PATCH / DELETE requests for Laravel adapter

### DISCUSSION THREAD

**Discussion by LucaRed** [inertiajs/inertia#469] [Ideas]
Hi everyone!

There is a [long-standing bug](https://bugs.php.net/bug.php?id=55815) in PHP which prevents submitted data from being read if the request method is PUT, PATCH or DELETE.

Laravel has a workaround for this: you make a normal POST request with a `_method` field containing the actual method name. [This comment](https://github.com/inertiajs/inertia-laravel/issues/180#issuecomment-719998370) explains it better than me.

I know that the main Inertia code is framework-agnostic, but it would be very useful if there was a way to globally configure Inertia to prevent real PUT requests and use the `_method` field instead.

**Comment by claudiodekker** [comment]
Thanks for adding this idea! You're absolutely right, that's 100% the reason we haven't implemented it yet. It's still good to have this idea tracked so we don't forget about it and can poll interest / discuss potential implementations it at the same time, though!

We actually considered adding 'mixins' of sorts at some point, but more or less decided not to implement them at the time (because depending on the frameworks, languages & their quirks, the scope of what these 'mixins' _should_ do could potentially vary _a lot_). Furthermore, you _really_ only need the `_method` field when using multipart/uploads, which we considered 'acceptable'.

Anyway, you've got my 👍 . It'd be perfect if we didn't have to think about this _at all_ anymore when using Inertia with Laravel.
